src/main/kotlin/no/skatteetaten/aurora/mokey/service/AuroraStatusCalculator.kt

```diff
@@ -126,4 +129,20 @@ class AuroraStatusCalculator {
 
         return ap.stream().anyMatch { p -> Instant.parse(p.openShiftPodExcerpt.startTime).isBefore(threshold) }
     }
+
+    private fun findMostCriticalStatus(healthStatuses: List<HealthStatusDetail>): AuroraStatus {
+
+        val getMostCriticalStatus = { acc: HealthStatusDetail, auroraStatus: HealthStatusDetail ->
+            if (auroraStatus.level.level > acc.level.level) auroraStatus else acc
+        }
+
+        val nonHealthyStatuses = healthStatuses.filter { it.level !== HEALTHY }
+
+        return when {
+            nonHealthyStatuses.isEmpty() -> AuroraStatus(HEALTHY, "", healthStatuses)
+            else -> nonHealthyStatuses.reduce(getMostCriticalStatus).let {
+                AuroraStatus(it.level, it.comment, healthStatuses)
```

Comment:
Bør comment være her? Akkruat slik det er nå så får man bare den første grunnen til at noe er i en status? Vil det ikke gi verdi at man får alle healthStatuses som har den mest alvorlige statusen og kun dem?

Burde denne hatt nonHealthyStatuses så man kan vise alle statuser som det er noe galt med?

Reply:
Tanken var at som oftest ved å fikse den mest kritiske statusen så vil de andre også bli fikset. De andre statusene kommer som en følge av den mest kritiske, slik jeg ser det. Det finnes enkelt tilfeller hvor det er to problemer, dette bør komme frem, men er ikke helt sikker på hvordan? Ved å fikse den ene så få feil på den andre kan være greit nok. De har alle de andre statusene som kan hjelpe dem i å fikse flere problemer på en gang.

Vi kan godt sende med nonHealthyStatuses, men jeg ville ta med de statusene fra pod'ene for å visualisere det i Konsollet. Kunne vært greit å vise at det faktisk har blitt gjort en helsesjekk mot applikasjonen. Vet ikke om dette er hensiktsmessig, men noe å vurdere.

Reply:
Tror vi må prate om denne over kode egentlig.